crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -306,6 +306,17 @@ impl<T: TransactionOrdering> TxPool<T> {
             .or_default()
             .update(on_chain_nonce, on_chain_balance);
 
+        let total_cost: U256 = self
+            .pending_transactions()
+            .into_iter()
+            .filter(|t| t.sender() == tx.sender() && t.nonce() != tx.nonce())
+            .map(|t| t.cost)
+            .sum();
```

Comment:
this is the wrong location and very inefficient here

this check should be applied when inserting the transaction, inside this loop:

https://github.com/paradigmxyz/reth/blob/3bba41a209e308abdbe8e88f601f365566befa7d/crates/transaction-pool/src/pool/txpool.rs#L1054-L1056

Reply:
@mattsse I've changed the location as requested:
- I've added a test `test_insert_overdraft` to confirm the overdraft error.
- I think that we have to modify the tests as well because if we stay with `let on_chain_balance = U256::ZERO;` then the tests will trigger the overdraft error each time, am I correct?